### PR TITLE
Remove dateparser as a dependency and use datetime objects in PrometheusConnect

### LINKED ISSUE #29: Issue with dateparser
https://github.com/AICoE/prometheus-api-client-python/blob/v0.0.2/prometheus_api_client/metric.py#L146-L148
Here, sometimes the dateparser parses date in a yyyy-dd-mm format which is incorrect.
We can add `settings={"DATE_ORDER": "YMD"}` to fix this issue.
While at it, we should make this change in every place dateparser.parse is used

### LINKED ISSUE #31: Metric Object oldest_data_datetime vague explanation
[Here](https://prometheus-api-client-python.readthedocs.io/en/v0.0.2/source/prometheus_api_client.html#prometheus_api_client.metric.Metric), the explanation for the `oldest_data_datetime` parameter is a little vague. It is also a little hard to parse strings and define datetime objects which are then used to truncate the dataframes.

I think we should just remove the whole string parsing monkey business from the `Metric` class (and from `PrometheusConnect` as well), instead what I propose is:
`oldest_data_datetime` can be a `datetime.datetime` object or a `datetime.timedelta` object passed as a parameter. 
* if `oldest_data_datetime` is an explicit `datetime.datetime` object, then any data older than it will be deleted.
* if `oldest_data_datetime` is a `datetime.timedelta` object, then any data that is older than [(latest metric timestamp) -`oldest_data_datetime`] will be deleted.

This will reduce the uncertainty for date parsing from inside the Metric class at least.

### PR DESCRIPTION
**Pull request by 4n4nd**
Use datetime objects for metric start_time and end_time.
Use timedelta objects for chunk_size.
Add tests for class PrometheusConnect
Move pretty_print_metric function to utils.py

Closes #29 
Closes #31

### CODE REVIEW

**Comment by durandom** [comment]
@4n4nd there are tests in this PR, but they're not run? didnt we have a travis integration?

**Comment by 4n4nd** [comment]
We do not have a travis integration. The tests need a promtheus instance to work so rn I run them locally

File: tests/test_prometheus_connect.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+class TestPrometheusConnect(unittest.TestCase):
+    """
+    Test module for class PrometheusConnect
+    """
+
+    def setUp(self):
+        """
+        set up connection settings for prometheus
+        """
+        self.prometheus_host = os.getenv("PROM_URL")
```
Comment (4n4nd) [inline tests/test_prometheus_connect.py]: @durandom This is where you need to specify the prometheus host url

**Comment by 4n4nd** [comment]
Fixed incorrect function use (we need `timedelta.total_seconds()` instead of `timedelta.seconds`)

**Comment by durandom** [comment]
@4n4nd think about how to add tests. Google stubs and mocks, ask frido for python specifics
